modules/claims_api/app/controllers/claims_api/v2/veterans/claims_controller.rb

```diff
@@ -105,7 +105,10 @@ def find_bgs_claim_in_lighthouse_collection(lighthouse_collection:, bgs_claim:)
         def find_lighthouse_claim!(claim_id:)
           lighthouse_claim = ClaimsApi::AutoEstablishedClaim.get_by_id_or_evss_id(claim_id)
 
-          if looking_for_lighthouse_claim?(claim_id: claim_id) && lighthouse_claim.blank?
+          claim_belongs_to_veteran = lighthouse_claim.veteran_icn == target_veteran.mpi.icn if lighthouse_claim.present?
+
+          if looking_for_lighthouse_claim?(claim_id: claim_id) &&
+             lighthouse_claim.blank? && claim_belongs_to_veteran
```

Comment:
I don't think this logic is quite right.

Let's say the use case is, "user is looking for a lighthouse claim but one doesn't exist".
In that scenario, `lighthouse_claim` will not be present, which means that `claim_belongs_to_veteran` is going to be `nil`.
That means that the expression on line#110 is going to see `true && true && nil`, which would cause the logic flow to not hit the `raise`.  (and we'd want it to raise because they were looking for a particular lighthouse claim, but one doesn't exist)

Reply:
TY